### ISSUE TITLE
Transect form -  Section number headers are not visible as the user scrolls down

### ISSUE THREAD

**Issue by andrewvanbreda** [BiologicalRecordsCentre/SPRING#69]
As you scroll down, the transect section numbers (headers) are not visible anymore, so mistakes are easily made because you have to scroll up and down all the time.

**Comment by andrewvanbreda** [comment]
Hi @BirenRathod,

I don't suppose you have any thoughts on this one, I am at a bit of a loss how to fix it.

If you go to this page (masquerade as me if needed)
https://test-eu-pollinator.pantheonsite.io/walk-data-entry?sample_id=19442009

Make sure the transect walk type is set to Butterfly.

If you click the "Next" button you will go to the species grid.

This is where there is this strange problem.

If you scroll down the species list, the header remains visible usually.
But sometimes if you reload the page it doesn't work.

(see these two screenshots, and note the header is not remaining sticky in one of them).

But I have no idea why this might happen.

The page uses the sticky-enabled class, which actually I think is a Drupal thing.

I can tell you it is not the theme, it does it even if we switch themes.

I notice the sticky-enabled class is still present on the table even when it isn't working.

Have you ever seen similar before? If not I will ask Gary.

Thanks for any thoughts

Andy

![Screen Shot 2022-08-01 at 12 07 22](https://user-images.githubusercontent.com/2463614/182143569-40877faa-0638-4735-a44b-118ecbe7cb90.png)

![Screen Shot 2022-08-01 at 12 06 25](https://user-images.githubusercontent.com/2463614/182143596-e5ed9bf3-8f20-4c8d-8856-1d749bbe6458.png)

**Comment by BirenRathod** [comment]
@andrewvanbreda The header only visible for first two columns. These two columns are '%sun' & '%cloud'. So I think not too much worry about it unless you found something different.

**Comment by andrewvanbreda** [comment]
Hi @BirenRathod 
Thanks for taking a looking, but yes it is different to this.
Sometimes when you scroll, the header is visible as you scroll the whole way down the table (which is what we want). Then other times you scroll and it vanishes after the first two attribute lines are scrolled.
I can't work out what would cause this behaviour as the sticky-enabled class is supposed to be an automatic Drupal thing. The only clue is that the table has 2 tbody areas, and it happens even if the theme is changed.
I do not know if EBMS has similar issue or if that is ok, I never managed to get that site to work.

**Comment by BirenRathod** [comment]
@andrewvanbreda Ok, In that case, I will a look bit more into details and let you know about this.

**Comment by DavidRoy** [comment]
@BirenRathod did you manage to resolve this?

**Comment by BirenRathod** [comment]
@DavidRoy It hasn't resolved yet, I'm still looking into this.

@andrewvanbreda The porblem is that the class name of species-grid of table header loaded twice which confused table to use sticky-header class in it. Why is table header loaded twice?

**Comment by andrewvanbreda** [comment]
@BirenRathod 
OK, leave this with me and I will see what is happening, I will let you know if I need further assistance.

I cannot comment this second, as the page is largely based on EBMS, so although I inherited/recoded the pages for EU PoMS, there is code in there I am not 100% familiar with.

Thanks for letting me know it is the class name loading twice causing the issue, that is very useful.

**Comment by BirenRathod** [comment]
@andrewvanbreda To understand bit more I have attached the screenshot here.
![table_screenshot](https://user-images.githubusercontent.com/5941255/193307148-8eeb6b48-a560-49c0-a533-fddae4e8bffa.png)

**Comment by andrewvanbreda** [comment]
@BirenRathod 
Thanks Biren.
Am pretty sure this is a bug in the code, that extra table (the top one of the two) is odd. I will let you know once investigated, I will need to fiddle with the configuration of the test site.
Looking at the code, it isn't obvious why that is appearing.

**Comment by andrewvanbreda** [comment]
Hi @BirenRathod 
Are you 100% sure that is the issue? I have tried lots of experiments this morning with no success.
I also see that EBMS has that extra grid also. 
I am still not sure where that extra hidden is coming from actually, it isn't in the prebuilt form code, it is being auto-generated elsewhere, but I suspect it is critical to the way the page works as EBMS has it.

Unfortunately, am having trouble testing the same scnario on EBMS because when I create tests transects on EBMS, they don't appear in my list (don't know if they need approval first).

Do we definetely know this section header problem doesn't occur on EMBS? It would be useful to know.

**Comment by BirenRathod** [comment]
@andrewvanbreda  Yes, I'm sure and that comes from this poms_sectioned_transects_input_sample.php

**Comment by andrewvanbreda** [comment]
Hi @BirenRathod 
I have tried to look at the again, but it is now talking more time than it is probably worth.

I fixed one minor problem, but the main problem is not the two grids.
When sticky-enabled is added to the table, that other table is added automatically containing the class sticky-header. 
I am not sure what creates it, I assume Drupal because from I can work out the "sticky-enabled" fucntionality is a Drupal thing.

That actually all works fine on the Bees/Hoverflies data entry tab which has those two grids. It is just the way it works.

The trouble is with the Butterflies tab which has sortable species names.
In the case the table has two tbody areas.
One containing the attributes rows %Cloud, % Sun at the top, the other containing the species.
These are needed so the table can just be sorted by species and can leave the attributes alone.

However the problem is having two tbody areas breaks the sticky header when I don't think it should.
I think there might be a core problem with sticky-enabled causing this.

The difficulty also being the form is based on EBMS, so I would need to completely rewrite the form two remove the tbody elements.

So at the moment I feel a bit stuck on this.
Let me know if you have any thoughts you could add that may help to resolve this situation.

**Comment by andrewvanbreda** [comment]
@Gary-van-Breda
Have you see this issue on EBMS? If not, do you know how to resolve it?
It is a form based on EBMS's forms where the table has attributes and species in it, but because it has 2 tbody sections, it seems to break stick-enabled.
If you scroll down beyond the first tbody, the stick-header vanishes.

An example page link is given above, but you would have to be registered masquerading as me to see it.
https://test-eu-pollinator.pantheonsite.io/walk-data-entry?sample_id=20472965
